drivers/mbox/mbox_nxp_imx_mu.c

```diff
@@ -40,13 +40,8 @@ static int nxp_imx_mu_send(const struct device *dev, uint32_t channel, const str
 		return -EINVAL;
 	}
 
-	/* Signalling mode. */
-	if (msg == NULL) {
-		return MU_TriggerInterrupts(cfg->base, kMU_GenInt0InterruptTrigger >> channel);
-	}
-
 	/* Data transfer mode. */
-	if (msg->size != MU_MBOX_SIZE) {
+	if ((msg->size < 0) || (msg->size > MU_MBOX_SIZE)) {
```

Comment:
if `msg` is null, won't you be referencing a nullptr here?

Reply:
Yes, you're right!
If `msg` is null we should return, as before, if `msg->data` is null we can send the message to host.
I'll update the PR.
Thanks!

Reply:
I'm not familiar with the MU HAL driver, so I may be wrong here. `MU_SendMsg` expects an `uint32_t` as msg payload and there is no argument to specify its length/size. Because `data32` variable is not zero-initialized, if you copy less than `sizeof(uint32_t)` bytes you may be sending garbage along with the data. How the receiver would know what is the useful part of the data payload sent?